node/actors/bft/src/leader/replica_prepare.rs

```diff
@@ -137,28 +137,38 @@ impl StateMachine {
             .or_insert_with(|| validator::PrepareQC::new(message.view.clone()))
             .add(&signed_message, self.config.genesis());
 
-        // We store the message in our cache.
-        self.prepare_message_cache
+        // Work on current view messages
+        let entry = self
```

Comment:
please consider fixing https://linear.app/matterlabs/issue/BFT-424/fix-aggregation-of-replicacommit-messages in the same pr.